HunterPie.Core/Networking/Http/HttpClientResponse.cs

```diff
@@ -70,10 +70,9 @@ public async Task DownloadAsync(string outPath)
 
         try
         {
-            if (!Directory.Exists(outPath))
-                Directory.CreateDirectory(
-                    Path.GetDirectoryName(outPath)!
-                );
+            string directoryPath = Path.GetDirectoryName(outPath)!;
+            if (!Directory.Exists(directoryPath))
+                Directory.CreateDirectory(directoryPath);
```

Comment:
This doesn't fix the issue you're reporting, the issue you reported is related with the creation of the file and happens when two or more threads are downloading the exact same file at the same time, which can happen when you access the Exported Hunts tab or when you're in a hunt with two monsters that have the exact same icon.

HunterPie is multithreaded, if two monsters of the same type spawn at the same time, it will create the ViewModels and the ViewModels will start a task to fetch the icon URI, if the URI is not found locally then it will hit the CDN to download the icon to the local folder and then return the local URI to that icon. The issue is that if both ViewModels start executing the fetch icon at the same time, they both will check the local path and see that the file is not there yet and will try to download the same icon, one of them will be able to finish first and the other will throw an exception saying that the file is already open by another process (in this case, HunterPie itself), it's a racing condition issue.

Adding a semaphore in this method would very likely fix the issue:
https://github.com/HunterPie/HunterPie/blob/660e2200d2e9c5f21900f007c1d310b4a4045451/HunterPie.Core/Remote/CDN.cs#L17-L43